spec/auth.md

```diff
@@ -0,0 +1,59 @@
+# Echo Server Authentication
+
+## Relay <--> Echo Server
+
+We use Ed25519 to verify that requests from the Relay network are valid and intended for the Echo Server Instance that has received them.
+
+### Implementation
+
+Echo Server generates<sup>[[1]](#generating-keys)</sup> an Ed25519 key-pair and stores that in-memory - or in files specified using environment variables. The generated
+public key is then sent to the Relay network along with a cloud app project id and the public url (both from environment variables) so that
+future requests to the relay can be signed and validated. The Echo Server instance also fetches the Relay network's public key and caches it
+so that requests received can be validated.
+
+On start-up Echo Server sends [the register request](#post-pushservers) to the relay network<sup>[[2]](#registering-with-relay)</sup>. This request is formatted using the
+public key from the above key-pair as well as the provided `project_id` and `public_url` (both provided via Environment Variables).
```

Comment:
We need to maintain consistency in the casing of the variables.

`project_id` should be `projectId`: https://github.com/WalletConnect/rs-relay/pull/354/files#diff-514195183eed7f300b70bc7617a8daf388bfdd6d01a52e3db2900aec6f66e4abR83

`public_url` is a new variable, but we should keep the same format of camelCase: `publicUrl`

Reply:
:+1: I will change it to use camelCase throughout

Reply:
It doesn't matter, but I think we should avoid fragmentation right now.

Reply:
I'm happy using camelCase I just defaulted to using underscores as I was unsure what is currently used throughout